[Jason Yap] iP

JasonYapzx opened the pull request:
## AI-lfred

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

AIlfred frees your mind of having to remember things you need to do. It's,


- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/JasonYapzx/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**!

Features:
- [x]  Managing tasks
- [x]  Managing deadlines (coming soon)
- [x]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) throws DukeException, IOException {
        Duke duke = new Duke();
        duke.initialize();
    }
}
```

Commit:
Add Gradle support

JasonYapzx committed:
Level-1

JasonYapzx committed:
Level-1

JasonYapzx committed:
Level-1

JasonYapzx committed:
Level 2

JasonYapzx committed:
Level 3

JasonYapzx committed:
Level 4

JasonYapzx committed:
A-TextUiTesting

JasonYapzx committed:
Level-5

JasonYapzx committed:
Level-6

JasonYapzx committed:
Level 6

JasonYapzx committed:
Added save functionality (Level 7)

JasonYapzx committed:
Added date parsing (Level 8)

JasonYapzx committed:
Merge branch 'branch-Level-7'

JasonYapzx committed:
Added date parsing (Level 8)

JasonYapzx committed:
Merge branch branch-Level 8

JasonYapzx committed:
Merge branch 'add-gradle-support'

JasonYapzx committed:
Added Gradle configs

JasonYapzx committed:
Added Storage class, skeleton files for Parser, TaskList and UI

JasonYapzx committed:
Added Parser, TaskList, UI functionalities. Abstracted Command Functions into Command Classes

JasonYapzx committed:
Refactored code into packages

JasonYapzx committed:
Added JUnit Tests for Storage, Deadline

JasonYapzx committed:
Refactored JavaDoc comments

JasonYapzx committed:
Refactored code to coding standards

JasonYapzx committed:
Added find functionality

JasonYapzx committed:
Merge branch 'branch-A-JavaDoc'

JasonYapzx committed:
Fixed merge conflicts

JasonYapzx committed:
Added find functionality

JasonYapzx committed:
Added Checkstyle files

JasonYapzx committed:
Followed checkstyle guidelines

JasonYapzx committed:
Merge branch 'A-CheckStyle'

gowribhat commented on src/main/java/duke/utils/Ui.java:
```diff
@@ -0,0 +1,45 @@
+
+
+    private static final String UI_GOODBYE = "               ☺ Saving your data before you go...\n"
+                                           + "               Bye. Hope to see you again soon!\n";
```
Good job in remembering to use screaming snake case for constants and using a common prefix for associated constants 👍🏽

gowribhat commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,135 @@
+                case("T"):
+                    task = new Todo(description);
+                    break;
+                case("D"):
```
Good job in ensuring correct indentation for switch-case 😄

gowribhat commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,135 @@
+        try (BufferedReader br = new BufferedReader(new FileReader(this.file))) {
+            String data = br.readLine();
+
+            while (data != null) {
```
White space used correctly 👍🏽

gowribhat commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,97 @@
+     * @return Date object used to construct Deadlines or Events.
+     * @throws DukeException Exception if date format is wrong.
+     */
+    public static Date parseDateSave(String input) throws DukeException {
```
I like your method names. Written in verb form and camelCase, it is easy to understand!

gowribhat reviewed (commented):
Overall looks good 😄 
Coding standards have been well implemented!

JasonYapzx committed:
Added JavaFX

wongyewjon commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,23 @@
+package duke.command;
```
Great use of separation of files into different packages.

wongyewjon reviewed (commented):
looks good, clean code

json-obj committed:
Fixed init issues

JasonYapzx committed:
Updated JavaFX UI/UX

JasonYapzx committed:
Merge branch 'branch-Level-10'

JasonYapzx committed:
Updated README

JasonYapzx committed:
Added META-INF file

JasonYapzx committed:
Added Assertions

JasonYapzx committed:
Refactored code quality

Presence of magic numbers within Command subclasses, getGreeting
function not abstracted

Easier understanding when reading and to follow the SLAP principle

Rename "magic numbers" with a constant, and replace magic numbers with
this constant on every occurrence, abstract getGreeting to UI class

Following the Code Quality guide

JasonYapzx committed:
Removed unnecessary imports

Presence of unused imports in Duke

In order to reduce LoC and follow coding standard

Removed imports from affected classes

-

JasonYapzx committed:
Merge pull request #1 from JasonYapzx/A-Assertions

Merge A-Assertions

JasonYapzx committed:
Merge pull request #2 from JasonYapzx/A-CodeQuality

Refactored code quality

JasonYapzx committed:
Added sort funtionality

JasonYapzx committed:
Refactored Javadocs

JasonYapzx committed:
Merge pull request #3 from JasonYapzx/branch-C-Sort

Merged branch-C-Sort

JasonYapzx committed:
Refactor code, Add Ui Screenshot

JasonYapzx committed:
Add User Guide

JasonYapzx committed:
Rename Ui PNG

JasonYapzx committed:
Rename Ui.png

JasonYapzx committed:
Fix UG
